src/mscorlib/src/System/Char.cs

```diff
@@ -234,7 +234,7 @@ public bool Equals(Char obj)
           if (IsLatin1(c)) {
               if (IsAscii(c)) {
```

Comment:
It really depends on what's more common generally. This way non-Latin1 characters benefit from being tested only once, Latin1 get tested twice. Put ASCII outside and ASCII chars get tested once, non-ASCII twice. Neither is clearly better from first principles.